tests/timeLockProtocolMiningReward.js

```diff
@@ -0,0 +1,1426 @@
+    'proposer trying to cancel already cancelled call should revert',
+    mochaAsync(async () => {
+      const id = await protocolMiningReward.hashOperation({
+        target, value, data, predecessor, salt,
+      });
+
+      await beproAssert.revertsFn(
+        () => protocolMiningReward.cancel({ id }),
+        'TimelockController: operation cannot be cancelled',
+      );
+    }),
+  );
+
+  it(
+    'proposer trying to cancel already executed/done scheduled call should revert',
+    mochaAsync(async () => {
+      getNextSalt();
+
+      /// NOTE: send eth to protocolMiningReward contract to test 'execute' function
+      const target = protocolMiningReward.getAddress();
+      const data = '0x';
+      const ethvalue = '1';
+      const ethvalueWei = web3utils.toWei(ethvalue, 'ether');
+      const value = ethvalueWei;
+
+      await protocolMiningReward.schedule({
+        target, value, data, predecessor, salt, delay, rewardTokenAmount,
+      });
+
+      const id = await protocolMiningReward.hashOperation({
+        target, value, data, predecessor, salt,
+      });
+
+      // do blockchain travel forward in time to execute scheduled call
+      await forwardTime(delay);
+      //
+
+      // execute scheduled call
+      await protocolMiningReward.execute({
+        target, value, data, predecessor, salt,
+      }, { value });
+
+      await beproAssert.revertsFn(
+        () => protocolMiningReward.cancel({ id }),
+        'TimelockController: operation cannot be cancelled',
+      );
+    }),
+  );
+
+  /// updateDelay
+  it(
+    'not-timelock trying to updateDelay should revert',
+    mochaAsync(async () => {
+      const newDelay = minDelay + 60; // 1 more minute for min delay
+
+      await beproAssert.revertsFn(
+        () => protocolMiningReward.updateDelay({ newDelay }),
+        'TimelockController: caller must be timelock',
+      );
+
+      await beproAssert.revertsFn(
+        () => protocolMiningReward.updateDelay({ newDelay }, { from: proposer }),
+        'TimelockController: caller must be timelock',
+      );
+
+      await beproAssert.revertsFn(
+        () => protocolMiningReward.updateDelay({ newDelay }, { from: executor }),
+        'TimelockController: caller must be timelock',
+      );
+    }),
+  );
+
+  // NOTE: the caller must be the timelock itself. This can only be achieved by scheduling and later executing
+  // an operation where the timelock is the target and the data is the ABI-encoded call to this function.
+  it(
+    'proposer schedules call for timelock contract itself to call updateDelay function and emit MinDelayChange event',
+    mochaAsync(async () => {
+      getNextSalt();
+
+      const newDelay = minDelay + 60; // 1 more minute for min delay
+      const target = protocolMiningReward.getAddress();
+      const data = encodeFuncSigWithParams(
+        'updateDelay(uint256)',
+        [ 'uint256' ],
+        [ newDelay ],
+      );
+
+      await protocolMiningReward.hashOperation({
```

Comment:
lingering `hashOperation` that's not assigned to anything, left over?

Reply:
right, code was cleaned and optimized.